spec/models/miq_queue_spec.rb

```diff
@@ -840,4 +841,15 @@ def self.some_method(single_arg)
       expect(MiqQueue.where(:id => q.id).count).to eq(0)
     end
   end
+
+  context "validates that the zone exists in the current region" do
+    it "with a matching region" do
+      zone = FactoryGirl.create(:zone)
+      expect(MiqQueue.create!(:state => "ready", :zone => zone.name)).to be_kind_of(MiqQueue)
+    end
+
+    it "without a matching region" do
+      expect { MiqQueue.create!(:state => "ready", :zone => "Missing Zone") }.to raise_error(ActiveRecord::RecordInvalid)
+    end
+  end
```

Comment:
@bdunne I glanced at the tests but didn't see one for a message with a nil zone... does that pass this validation and not raise an error?

Reply:
:+1: Done